src/configuration.rs

```diff
@@ -38,6 +38,22 @@ pub struct RateLimitPolicy {
 }
 
 impl RateLimitPolicy {
+    pub fn new(
+        hosts: GlobPatternSet,
+        rules: Option<Vec<Rule>>,
+        global_actions: Option<Vec<RLA_action_specifier>>,
+        upstream_cluster: Option<String>,
+        domain: Option<String>,
```

Comment:
Unsure about these `Option`s here, but I guess we'll see how this pans out.

Reply:
Without `Option`, you make a strict requirement for that field to be present else, it will fail the shim during loading, and the control plane has no option to get that info. Thus, always avoid irrecoverable failure in wasm :)

Reply:
yeah, a default is a much better choice like `NO_DOMAIN_FOUND`.

Reply:
a default domain can lead to unexpected rate limit behavior. That's why I tend to think that it should be required and fail if not present.

We can leave as it is for now.